src/python/pants/scm/subsystems/changed.py

```diff
@@ -160,10 +160,35 @@ def register_options(cls, register):
         )
 
 
+# TODO: This rule is uncacheable due to using an unsandboxed process execution. @_uncacheable_rule
+# is not exposed to the public API, and it would be preferable to focus the "uncacheability" to each
+# individual git execution rather than this entire rule.
+@_uncacheable_rule
+def changed_files(request: ChangedFilesRequest) -> ChangedFiles:
```

Comment:
This is pretty cool that it's now a rule!

In the future, I think your TODO is saying that we should refactor `git` to not use `subprocess` anymore, right? To confirm, it's safe for now to use `subprocess` because of `@_uncacheable_rule`?